common/surface.c

```diff
@@ -891,7 +891,7 @@ l_surface_getRawPixel(lua_State *L)
 	int x = luaL_checkinteger(L, 2);
 	int y = luaL_checkinteger(L, 3);
 	int size = surf->format->BytesPerPixel;
-	Uint8 *ptr = (Uint8 *)surf->pixels + y * surf->pitch + x * size;
+	char *ptr = (char *)surf->pixels + y * surf->pitch + x * size;
 	lua_pushlstring(L, ptr, size);
```

Comment:
Is the warning still suppressed if you instead do:

```c
lua_pushlstring(L, (char *) ptr, size);
```
?

That might better communicate "this is an unsigned value but the API demands we call it a char"